front/src/common/Map/Layers/blankStyle.ts

```diff
@@ -35,7 +69,7 @@ export const useMapBlankStyle = (): MapProps['mapStyle'] => {
         },
       ],
     } as MapProps['mapStyle'];
-  }, [signalingSystems]);
+  }, [validSprites, signalingSystems]);
```

Comment:
I think you can keep only this one, as validSprites is already updated when signaling systems changes
```suggestion
  }, [validSprites]);
```